client/packages/common/src/ui/components/inputs/TextInput/NumericTextInput.tsx

```diff
@@ -164,7 +167,8 @@ export const NumericTextInput: FC<NumericTextInputProps> = React.forwardRef(
           onChange(newNum);
         }}
         onBlur={() => {
-          onChange(Number(parse(textValue ?? '')) || undefined);
+          const parsed = parse(textValue ?? '');
+          onChange(Number.isNaN(parsed) ? undefined : parsed);
```

Comment:
My fix the other day (changing `||` to `??`) caused a different problem instead (showed `NaN`). Tests will now prevent this.